### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by Drew-wp**
Some more documentation for cameras and device inputs.

### CODE REVIEW

File: gui/src/advancedDynamicTexture.ts [hunk 1]
```diff
@@ -529,6 +529,9 @@ module BABYLON.GUI {
             return result;
         }
 
+		/**
+		 * FullScreenUI is created in a layer. This allows texture.layer.layerMask to be set.
```
Comment (RaananW) [inline gui/src/advancedDynamicTexture.ts]: This requires a lot more explanation than this sentence.
Also, would be great to have the parameters documented.

File: src/Cameras/babylon.camera.ts [hunk 2]
```diff
@@ -78,6 +78,10 @@
         @serializeAsVector3()
         public position: Vector3;
 
+		/**
+		 * default : Vector3(0.0, 0.1, 0.0) Y+ is up.
```
Comment (RaananW) [inline src/Cameras/babylon.camera.ts]: default up is 0,1,0 , not 0.1

File: src/Cameras/babylon.cameraInputsManager.ts [hunk 3]
```diff
@@ -28,6 +28,10 @@ module BABYLON {
             this.checkInputs = () => { };
         }
 
+		/**
+		 * Add an input method to a camera
+		 * @param input
```
Comment (RaananW) [inline src/Cameras/babylon.cameraInputsManager.ts]: not sure "add" and "remove" need documentation (kind'a like a getter and a setter), BUT - if you already added it, please document the parameter as well.

File: src/Cameras/babylon.deviceOrientationCamera.ts [hunk 4]
```diff
@@ -23,6 +28,9 @@ module BABYLON {
             }
         }
 
+		/**
+		 * Rotates the camera to 0Deg around the axis passed. If Axis.Y is passed, the camera will be moved around Y to face z+.
```
Comment (RaananW) [inline src/Cameras/babylon.deviceOrientationCamera.ts]: The parameter documentation could be the 2nd sentence.

File: src/Cameras/babylon.freeCamera.ts [hunk 5]
```diff
@@ -15,14 +15,20 @@
         public inputs: FreeCameraInputsManager;
 
         //-- begin properties for backward compatibility for inputs
-        public get angularSensibility(): number {
+		/**
+		 * This setting affects input for the mouse.
```
Comment (RaananW) [inline src/Cameras/babylon.freeCamera.ts]: missing information as to how. Same goes to the setter (the parameter documentation would be the most important)

File: gui/src/advancedDynamicTexture.ts (hunk 1 above)
Comment (Drew-wp) [reply gui/src/advancedDynamicTexture.ts]: I'll add as much as I understand. That may be a shockingly small amount though. So far I have been trying to add the things I wish I had been able to find while working on my project.

**Comment by Drew-wp** [comment]
I have changed all my indenting to spaces. sorry about that. I also corrected the mistake in the default camera orientation i made.

File: gui/src/advancedDynamicTexture.ts [hunk 6]
```diff
@@ -529,6 +529,12 @@ module BABYLON.GUI {
             return result;
         }
 
+        /**
+         * FullScreenUI is created in a layer. This allows it to be treated like any other layer.
+         * As such, if you have a multi camera setup, you can set the layerMask on the GUI as well.
+         * When the GUI is not Created as FullscreenUI it does not respect the layerMask.
+         * layerMask is set through advancedTexture.layer.layerMask
+         */
```
Comment (deltakosh) [inline gui/src/advancedDynamicTexture.ts]: Parameters need to be documented alongside return value

File: src/Cameras/babylon.camera.ts [hunk 7]
```diff
@@ -78,6 +78,10 @@
         @serializeAsVector3()
         public position: Vector3;
 
+        /**
+         * default : Vector3(0, 1, 0) Y+ is up.
```
Comment (deltakosh) [inline src/Cameras/babylon.camera.ts]: Please add a small line (even an obvious one like gets or sets the up vector used to define where the up is

File: src/Cameras/babylon.camera.ts [hunk 8]
```diff
@@ -116,16 +120,16 @@
         public viewport = new Viewport(0, 0, 1.0, 1.0);
 
         /**
-        * Restricts the camera to viewing objects with the same layerMask.
-        * A camera with a layerMask of 1 will render meshes with no layerMask and meshes with a layerMask of 1.
-        */
+         * Restricts the camera to viewing objects with the same layerMask.
+         * A camera with a layerMask of 1 will render meshes with no layerMask and meshes with a layerMask of 1.
```
Comment (deltakosh) [inline src/Cameras/babylon.camera.ts]: Actually the camera will display all meshes where mesh.layerMask & camera.layerMask!== 0

File: src/Cameras/babylon.camera.ts [hunk 9]
```diff
@@ -116,16 +120,16 @@
...
+         */
         @serialize()
         public layerMask: number = 0x0FFFFFFF;
 
         /**
-        * default : FOVMODE_VERTICAL_FIXED
-        * fovMode sets the camera frustum bounds to the viewport bounds.
-        */
+         * default : FOVMODE_VERTICAL_FIXED
```
Comment (deltakosh) [inline src/Cameras/babylon.camera.ts]: Please add default as blah (default is xxx) format

File: src/Cameras/babylon.cameraInputsManager.ts [hunk 10]
```diff
@@ -28,6 +28,12 @@ module BABYLON {
             this.checkInputs = () => { };
         }
 
+        /**
+         * Add an input method to a camera.
+         * builtin inputs example: camera.inputs.addGamepad();
+         * custom inputs example: camera.inputs.add(new BABYLON.FreeCameraGamepadInput());
+         * @param input
```
Comment (deltakosh) [inline src/Cameras/babylon.cameraInputsManager.ts]: Missing input descy

File: src/Cameras/babylon.freeCamera.ts [hunk 11]
```diff
@@ -15,6 +15,7 @@
         public inputs: FreeCameraInputsManager;
 
         //-- begin properties for backward compatibility for inputs
+
         public get angularSensibility(): number {
```
Comment (deltakosh) [inline src/Cameras/babylon.freeCamera.ts]: Should be in for of gets or sets a value blah blah

**Comment by deltakosh** [comment]
Great! looks good to me :)

**Pull request merged**